tests/resources/test_resources_etag.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Service tests."""
+
+import json
+
+import pytest
+from invenio_search import current_search, current_search_client
+
+
+@pytest.fixture()
+def input_data():
+    """Input data (as coming from the view layer)."""
+    return {
+        'metadata': {
+            'title': 'Test'
+        },
+    }
+
+
+@pytest.fixture()
+def headers():
+    """Default headers for making requests."""
+    return {
+        'content-type': 'application/json',
+        'accept': 'application/json',
+    }
+
+
+def test_etag_update(app, client, input_data, headers):
+    """Test a simple REST API flow."""
+    h = headers
+
+    # Create a record
+    res = client.post('/mocks', headers=h, data=json.dumps(input_data))
+    assert res.status_code == 201
+    id_ = res.json['id']
+    assert res.json['metadata'] == input_data['metadata']
+
+    revision_id = res.json["revision_id"]
+
+    # Update headers
+    h.update(dict(if_match=revision_id))
```

Comment:
Minor: There's code duplication between the two tests. This part can be extracted as a fixture.